qiskit/primitives/backend_estimator_v2.py

```diff
@@ -0,0 +1,264 @@
+
+    This class does not perform any measurement or gate mitigation, and, presently, is only
+    compatible with Pauli-based observables.
+
+    Each tuple of ``(circuit, observables, <optional> parameter values, <optional> precision)``,
+    called an estimator primitive unified bloc (PUB), produces its own array-based result. The
+    :meth:`~.BackendEstimatorV2.run` method can be given a sequence of pubs to run in one call.
+    """
+
+    def __init__(
+        self,
+        *,
+        backend: BackendV1 | BackendV2,
+        default_precision: float = 0.015625,
+        abelian_grouping: bool = True,
+    ):
+        """
+        Args:
+            backend: The backend to run the primitive on.
+            default_precision: The default precision to use if none are specified in :meth:`~run`.
+                Default: 0.015625 (1 / sqrt(4096)).
+            abelian_grouping: Whether the observables should be grouped into sets of
+                qubit-wise commuting observables.
+        """
+        super().__init__()
+        self._backend = backend
+        self._default_precision = default_precision
+        self._abelian_grouping = abelian_grouping
+
+        basis = PassManagerConfig.from_backend(backend).basis_gates
+        if isinstance(backend, BackendV2):
+            opt1q = Optimize1qGatesDecomposition(basis=basis, target=backend.target)
+        else:
+            opt1q = Optimize1qGatesDecomposition(basis=basis)
+        self._passmanager = PassManager([opt1q])
+
+    @property
+    def default_precision(self) -> float:
+        """Return the default precision"""
+        return self._default_precision
+
+    @property
+    def ablian_grouping(self) -> bool:
+        """Return whether Abelian grouping is used or not."""
+        return self._abelian_grouping
+
+    @property
+    def backend(self) -> BackendV1 | BackendV2:
+        """Returns the backend which this sampler object based on."""
+        return self._backend
+
+    def run(
+        self, pubs: Iterable[EstimatorPubLike], *, precision: float | None = None
+    ) -> PrimitiveJob[PrimitiveResult[PubResult]]:
+        if precision is None:
+            precision = self._default_precision
+        coerced_pubs = [EstimatorPub.coerce(pub, precision) for pub in pubs]
+        self._validate_pubs(coerced_pubs)
+        job = PrimitiveJob(self._run, coerced_pubs)
+        job._submit()
+        return job
+
+    def _validate_pubs(self, pubs: list[EstimatorPub]):
+        for i, pub in enumerate(pubs):
+            if pub.precision == 0.0:
+                raise ValueError(
+                    f"The {i}-th pub has precision 0. But precision should be larger than 0."
+                )
+
+    def _run(self, pubs: list[EstimatorPub]) -> PrimitiveResult[PubResult]:
+        return PrimitiveResult([self._run_pub(pub) for pub in pubs])
+
+    def _run_pub(self, pub: EstimatorPub) -> PubResult:
+        shots = int(np.ceil(1.0 / pub.precision**2))
```

Comment:
I removed `_VARIANCE_UPPER_BOUND` and simplified this part since the numerator is 1.